apps/language_server/lib/language_server/providers/completion.ex

```diff
@@ -593,6 +601,51 @@ defmodule ElixirLS.LanguageServer.Providers.Completion do
     end
   end
 
+  def suggest_module_name(file_path) when is_binary(file_path) do
+    file_path
+    |> Path.split()
+    |> Enum.reverse()
+    |> do_suggest_module_name()
+  end
+
+  defp do_suggest_module_name([]), do: nil
+
+  defp do_suggest_module_name([filename | reversed_path]) do
+    filename
+    |> String.split(".")
+    |> case do
+      [file, "ex"] ->
+        do_suggest_module_name(reversed_path, [file], topmost_parent: "lib")
+
+      [file, "exs"] ->
+        if String.ends_with?(file, "_test") do
+          do_suggest_module_name(reversed_path, [file], topmost_parent: "test")
+        else
+          nil
+        end
+
+      _otherwise ->
+        nil
+    end
+  end
+
+  defp do_suggest_module_name([dir | _rest], module_name_acc, topmost_parent: topmost)
+       when dir == topmost do
+    module_name_acc
+    |> Enum.map(&Macro.camelize/1)
+    |> Enum.join(".")
+  end
+
+  defp do_suggest_module_name([dir_name | rest], module_name_acc, opts) do
+    do_suggest_module_name(rest, [dir_name | module_name_acc], opts)
+  end
+
+  defp do_suggest_module_name([], _module_name_acc, _opts) do
```

Comment:
is it optimal to go all the way up to `/` or `C:\`? We could stop at workdir level.